src/components/SearchMultipleSelectionPicker.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useLocalize from '@hooks/useLocalize';
+import localeCompare from '@libs/LocaleCompare';
+import Navigation from '@libs/Navigation/Navigation';
+import type {OptionData} from '@libs/ReportUtils';
+import ROUTES from '@src/ROUTES';
+import Button from './Button';
+import SelectionList from './SelectionList';
+import SelectableListItem from './SelectionList/SelectableListItem';
+
+type SearchMultipleSelectionPickerItem = {
+    name: string;
+    value: string;
+};
+
+type SearchMultipleSelectionPickerProps = {
+    items: SearchMultipleSelectionPickerItem[];
+    initiallySelectedItems: SearchMultipleSelectionPickerItem[] | undefined;
+    pickerTitle?: string;
+    onSaveSelection: (values: string[]) => void;
+    shouldShowTextInput?: boolean;
+};
+
+function SearchMultipleSelectionPicker({items, initiallySelectedItems, pickerTitle, onSaveSelection, shouldShowTextInput = true}: SearchMultipleSelectionPickerProps) {
```

Comment:
@SzymczakJ can you please move this file inside `/components/Search`? this directory already exists and let's group all reusable Search related components there